packages/admin/admin/src/form/FinalFormFileUpload.tsx

```diff
@@ -0,0 +1,164 @@
+import { Delete, Select } from "@comet/admin-icons";
+import { Button, Chip, ComponentsOverrides, FormHelperText, IconButton, Theme } from "@mui/material";
+import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import clsx from "clsx";
+import * as React from "react";
+import { Accept, useDropzone } from "react-dropzone";
+import { FieldRenderProps } from "react-final-form";
+import { FormattedMessage } from "react-intl";
+
+import { PrettyBytes } from "../helpers/PrettyBytes";
+
+export type FinalFormFileUploadClassKey = "root" | "droppableArea" | "droppableAreaCaption" | "fileList" | "fileListItem" | "disabled" | "error";
+
+const styles = ({ palette }: Theme) => {
+    return createStyles<FinalFormFileUploadClassKey, FinalFormFileUploadProps>({
+        root: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            gap: "10px",
+            alignSelf: "stretch",
+            minWidth: "350px",
+        },
+        droppableArea: {
+            display: "flex",
+            height: "80px",
+            padding: "10px",
+            justifyContent: "center",
+            alignItems: "center",
+            alignSelf: "stretch",
+            borderRadius: "4px",
+            border: `1px dashed ${palette.text.secondary}`,
+            cursor: "pointer",
+            minWidth: "250px",
+            "&$disabled": {
+                cursor: "default",
+            },
+            "&$error": {
+                border: `1px dashed ${palette.error.main}`,
+            },
+        },
+        droppableAreaCaption: {
+            color: palette.text.secondary,
+            fontSize: "14px",
+            fontStyle: "normal",
+            fontWeight: "300",
+            lineHeight: "20px",
+        },
+        fileList: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            gap: "2px",
+            alignSelf: "stretch",
+        },
+        fileListItem: {
+            display: "flex",
+            height: "35px",
+            padding: "8px 7px 8px 15px",
+            alignItems: "center",
+            gap: "10px",
+            alignSelf: "stretch",
+            borderRadius: "4px",
+            background: palette.background.default,
+            justifyContent: "space-between",
+            minWidth: "250px",
+        },
+        disabled: {},
+        error: {},
+    });
+};
+
+export interface FinalFormFileUploadProps extends FieldRenderProps<File[], HTMLInputElement> {
+    dropzoneVariant: "dropzoneOnly" | "buttonOnly" | "default";
+    caption: React.ReactNode;
+    multipleFiles: boolean;
```

Comment:
multiple is already a part of FieldInputProps, I can use it but then I don't need to declare it in FinalFormFileUploadProps at all, because it extends FieldRenderProps.